contracts/libraries/VaultLifecycle.sol

```diff
@@ -33,6 +33,9 @@ library VaultLifecycle {
     }
 
     function commitAndClose(
+        address strikeSelection,
+        address optionsPremiumPricer,
+        uint256 premiumDiscount,
```

Comment:
Isn't this is a security risk because callers can pass in any strike selection / pricer contract they want? We should make these contracts fixed